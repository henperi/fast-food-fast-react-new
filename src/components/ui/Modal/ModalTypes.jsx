import React from 'react';

export const AddToCartModal = (props) => {
  const {
    title, price, description, handleModal,
  } = props;
  console.log('insideAddToCart', props);
  return (
    <div>
      <h2 className="text-center">Add this food item to existing Cart</h2>
      <div className="content-div">
        <div className="item-title">
          {title}
-
          <span className="badge price">
            &#8358;
            {price}
          </span>
        </div>
        <div className="item-description">
          <p>{description}</p>
        </div>
      </div>

      <button type="button" className="btn btn-blue btn-block btn-rounded btn-bg">
        Add To Cart
      </button>

      <button
        onClick={handleModal}
        type="button"
        className="close-button btn btn-primary btn-block btn-rounded btn-bg"
      >
        Cancel
      </button>
    </div>
  );
};

export const RemoveFromCartModal = (props) => {
  const {
    title, price, description, handleModal,
  } = props;

  return (
    <div>
      <h2 className="text-center">Remove this food item from existing Cart</h2>
      <div className="content-div">
        <div className="item-title">
          {title}
-
          <span className="badge price">
            &#8358;
            {price}
          </span>
        </div>
        <div className="item-description">
          <p>{description}</p>
        </div>
      </div>

      <button type="button" className="btn btn-red btn-block btn-rounded btn-bg">
        Remove From Cart
      </button>

      <button
        onClick={handleModal}
        type="button"
        className="close-button btn btn-primary btn-block btn-rounded btn-bg"
      >
        Cancel
      </button>
    </div>
  );
};

export const SignUpModal = (props) => {
  const { handleModal } = props;

  return (
    <div>
      <h2 className="text-center">Signup Now</h2>
      <div className="content-div">
        <div className="item-description">
          <input />
        </div>
      </div>
      <span>
        <button type="button" className="btn btn-green btn-blockX btn-rounded btn-bg">
          Sign Up
        </button>

        <button
          onClick={handleModal}
          type="button"
          className="close-button btn btn-default btn-blockX btn-rounded btn-bg"
        >
          Cancel
        </button>
      </span>
    </div>
  );
};
