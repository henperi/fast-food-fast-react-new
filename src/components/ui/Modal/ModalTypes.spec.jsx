import React from 'react';
import { shallow } from 'enzyme';
import { AddToCartModal, RemoveFromCartModal, mapStateToProps } from './ModalTypes';

test('should render ModalTypes correctly', () => {
  const wrapper = shallow(<AddToCartModal />);
  expect(wrapper).toMatchSnapshot();
});
test('should render ModalTypes correctly', () => {
  const wrapper = shallow(<RemoveFromCartModal />);
  expect(wrapper).toMatchSnapshot();
});

/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
// import { shallow } from 'enzyme';
// import { PlaceOrderModal } from './PlaceOrderModal';

describe('rendering', () => {
  let wrapper;
  let props;
  // let mockedSignUpState;
  beforeEach(() => {
    props = {
      handleModal: jest.fn(),
      handleAddToCart: jest.fn(),
      removeFromCart: jest.fn(),
      cart: [],
      placeOrder: jest.fn(),
    };
  });

  it('should render AddToCartComponent the component', () => {
    wrapper = shallow(<AddToCartModal {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  it('should test the handleAddToCart btn function click', () => {
    const fakeEvent = { preventDefault: () => 'preventDefault' };
    wrapper = shallow(<AddToCartModal {...props} />);
    const addToCart = wrapper.find('.btn-blue');
    addToCart.simulate('click', fakeEvent);
    const handleAddToCart = jest.fn(props.cart);
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().onSubmit);
    });
    promise.then(() => expect(handleAddToCart).toHaveBeenCalledTimes(1));
  });

  it('should test mapStateToProps', () => {
    const state = mapStateToProps(props);
    expect(state).toBeTruthy();
  });
});
