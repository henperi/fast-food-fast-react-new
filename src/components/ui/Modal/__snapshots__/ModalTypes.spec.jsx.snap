// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render AddToCartComponent the component 1`] = `
<div>
  <h2
    className="text-center"
  >
    Add this food item to existing Cart
  </h2>
  <div
    className="content-div"
  >
    <div
      className="item-title"
    >
      <span
        className="badge price"
      >
        ₦
      </span>
    </div>
    <div
      className="item-description"
    >
      <p />
    </div>
  </div>
  <button
    className="btn btn-blue btn-block btn-rounded btn-bg"
    onClick={[Function]}
    type="button"
  >
    Add To Cart
  </button>
  <button
    className="close-button btn btn-primary btn-block btn-rounded btn-bg"
    onClick={[MockFunction]}
    type="button"
  >
    Cancel
  </button>
</div>
`;

exports[`should render ModalTypes correctly 1`] = `
<div>
  <h2
    className="text-center"
  >
    Add this food item to existing Cart
  </h2>
  <div
    className="content-div"
  >
    <div
      className="item-title"
    >
      <span
        className="badge price"
      >
        ₦
      </span>
    </div>
    <div
      className="item-description"
    >
      <p />
    </div>
  </div>
  <button
    className="btn btn-blue btn-block btn-rounded btn-bg"
    onClick={[Function]}
    type="button"
  >
    Add To Cart
  </button>
  <button
    className="close-button btn btn-primary btn-block btn-rounded btn-bg"
    type="button"
  >
    Cancel
  </button>
</div>
`;

exports[`should render ModalTypes correctly 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
